pkg/adapter/nimbus-k8tls/manager/manager.go

```diff
@@ -0,0 +1,176 @@
+//+kubebuilder:rbac:groups=intent.security.nimbus.com,resources=clusternimbuspolicies/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=batch,resources=cronjobs,verbs=get;create;delete;list;watch;update
+
+func Run(ctx context.Context) {
+	cwnpCh := make(chan string)
+	deletedcwnpCh := make(chan *unstructured.Unstructured)
+	go globalwatcher.WatchClusterNimbusPolicies(ctx, cwnpCh, deletedcwnpCh)
+
+	updateCronJobCh := make(chan common.Request)
+	deletedCronJobCh := make(chan common.Request)
+	go watcher.WatchCronJobs(ctx, updateCronJobCh, deletedCronJobCh)
+
+	for {
+		select {
+		case <-ctx.Done():
+			close(cwnpCh)
+			close(deletedcwnpCh)
+
+			close(updateCronJobCh)
+			close(deletedCronJobCh)
+			return
+
+		case createdCwnp := <-cwnpCh:
+			createOrUpdateCronJob(ctx, createdCwnp)
+		case deletedCwnp := <-deletedcwnpCh:
+			logCronJobsToDelete(ctx, deletedCwnp)
+
+		case updatedCronJob := <-updateCronJobCh:
+			reconcileCronJob(ctx, updatedCronJob.Name)
+		case deletedCronJob := <-deletedCronJobCh:
+			reconcileCronJob(ctx, deletedCronJob.Name)
+		}
+	}
+}
+
+func reconcileCronJob(ctx context.Context, name string) {
+	logger := log.FromContext(ctx)
+	cwnpName := adapterutil.ExtractAnyNimbusPolicyName(name)
+	var cwnp v1alpha1.ClusterNimbusPolicy
+	err := k8sClient.Get(ctx, types.NamespacedName{Name: cwnpName}, &cwnp)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			logger.Error(err, "failed to get ClusterNimbusPolicy", "ClusterNimbusPolicy.Name", cwnpName)
+		}
+		return
+	}
+	createOrUpdateCronJob(ctx, cwnpName)
+}
+
+func createOrUpdateCronJob(ctx context.Context, cwnpName string) {
+	logger := log.FromContext(ctx)
+	var cwnp v1alpha1.ClusterNimbusPolicy
+	if err := k8sClient.Get(ctx, types.NamespacedName{Name: cwnpName}, &cwnp); err != nil {
+		logger.Error(err, "failed to get ClusterNimbusPolicy", "ClusterNimbusPolicy.Name", cwnpName)
+		return
+	}
+
+	if adapterutil.IsOrphan(cwnp.GetOwnerReferences(), "ClusterSecurityIntentBinding") {
+		logger.V(4).Info("Ignoring orphan ClusterNimbusPolicy", "ClusterNimbusPolicy.Name", cwnpName)
+		return
+	}
+
+	deleteDanglingCj(ctx, logger, cwnp)
+	cronJob, configMap := builder.BuildCronJob(ctx, cwnp)
+
+	if cronJob != nil {
+		if err := setupK8tlsEnv(ctx, cwnp, scheme, k8sClient); err != nil {
+			logger.Error(err, "failed to setup k8tls env")
+			return
+		}
+		if configMap != nil {
+			if err := createCm(ctx, cwnp, scheme, k8sClient, configMap); err != nil {
+				logger.Error(err, "failed to create ConfigMap", "ConfigMap.Name", configMap.Name)
+				return
+			}
+		}
+		createOrUpdateCj(ctx, logger, cwnp, cronJob)
+	}
+}
+
+func logCronJobsToDelete(ctx context.Context, deletedCwnp *unstructured.Unstructured) {
+	logger := log.FromContext(ctx)
+
+	var existingCronJobs batchv1.CronJobList
+	if err := k8sClient.List(ctx, &existingCronJobs, &client.ListOptions{Namespace: NamespaceName}); err != nil {
```

Comment:
I am assuming that you are printing all the cronJObs in the namespace for debugging purpose..

Reply:
Nope, this message is to inform end user that certain CronJobs were deleted due to deletion of x CSIB.